fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/utils/SQLBuilder.java

```diff
@@ -118,9 +121,40 @@ public String getSQLTemplate() {
     /*
      * Returns a String representation suitable for debugging and log output.
      * This is ONLY intended for debugging in logs, and NEVER for passing to a JDBC database.
+     */
     @Override
     public String toString() {
-        return "SQLBuilder{..."; // TODO implement this...
+       StringBuilder whereClause  = new StringBuilder("SQLBuilder{");
+       for (int i=0;i<args.size();i++)
+        {
+            if (i!=0)
+            {
+                whereClause.append("  AND  ");
+            }
+            else
+            {
+                whereClause.append("WHERE  ");
+            }
+            Object currentArg = args.get(i);
+            String currentCri = crts.get(i);
+            StringBuilder modifiedArg  = new StringBuilder("[");
```

Comment:
I think you could still save this `StringBuilder` and just `whereClause.append(...)` as you go? Just a thought.

Reply:
Yes, on it.